src/LmdbCache.ts

```diff
@@ -73,8 +76,35 @@ export class LmdbCache<V = any> implements SortKeyCache<V> {
     }
   }
 
-  async put(stateCacheKey: CacheKey, value: V): Promise<void> {
-    await this.db.put(`${stateCacheKey.contractTxId}|${stateCacheKey.sortKey}`, value);
+  async put(cacheKey: CacheKey, value: V): Promise<void> {
+    await this.db.transaction(() => {
+      this.db.putSync(`${cacheKey.contractTxId}|${cacheKey.sortKey}`, value);
+
+      // Make sure there isn't too many entries for one contract
+      const numInCache = 1 + this.db.getKeysCount({
```

Comment:
not sure I understand why the `1 +`...

Reply:
getKeysCount doesn't catch the putSync that is just abobe, I'll add a comment